WordPress/src/main/java/org/wordpress/android/ui/people/RoleChangeDialogFragment.java

```diff
@@ -0,0 +1,151 @@
+package org.wordpress.android.ui.people;
+
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.RadioButton;
+import android.widget.TextView;
+
+import org.wordpress.android.R;
+
+public class RoleChangeDialogFragment extends DialogFragment {
+    private static final String PERSON_ID_TAG = "person_id";
+    private static final String PERSON_LOCAL_TABLE_BLOG_ID_TAG = "local_table_blog_id";
+    private static final String ROLE_TAG = "role";
+
+    private RoleListAdapter mRoleListAdapter;
+    private OnChangeListener mListener;
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        String role = mRoleListAdapter.getSelectedRole();
+        outState.putString(ROLE_TAG, role);
+    }
+
+    public static RoleChangeDialogFragment newInstance(long personID, int localTableBlogId, String role) {
+        RoleChangeDialogFragment roleChangeDialogFragment = new RoleChangeDialogFragment();
+        Bundle args = new Bundle();
+
+        args.putLong(PERSON_ID_TAG, personID);
+        args.putInt(PERSON_LOCAL_TABLE_BLOG_ID_TAG, localTableBlogId);
+        if (role != null) {
+            args.putString(ROLE_TAG, role);
+        }
+
+        roleChangeDialogFragment.setArguments(args);
+        return roleChangeDialogFragment;
+    }
+
+    public void setOnChangeListener(OnChangeListener listener) {
```

Comment:
Hey @oguzkocer , I'm pretty sure that this will introduce rotation problems, where the listener will just end up being null because the caller did not set the listener again.